tol_lab_share/messages/consumed/base_validator.py

```diff
@@ -0,0 +1,60 @@
+from abc import ABC, abstractmethod
+import logging
+from uuid import UUID
+
+from tol_lab_share import error_codes
+from tol_lab_share.error_codes import ErrorCode
+
+from .message_field import MessageField
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BaseValidator(ABC):
+    def __init__(self) -> None:
+        super().__init__()
+        self._errors: list[ErrorCode] = []
+
+    @abstractmethod
+    def validate(self) -> bool: ...
+
+    @property
+    def errors(self) -> list[ErrorCode]:
+        """Gets the list of errors that occurred during validation.
+
+        Returns:
+            list[ErrorCode]: A list of errors that occurred during validation.
+        """
+        return self._errors.copy()
+
+    def _check_is_uuid(self, field: MessageField) -> bool:
```

Comment:
A minor thing: is it better to generalise (as it is the base validator, which is a general validator inherited by the concrete validators) this function by letting the client decide which UUID version the validator is going to validate the `field` against?

Not sure whether if it's too much work for this though.

Reply:
I think I'm going to refer to YAGNI for this.  (You Ain't Gonna Need It).  We can provide ultimate flexibility in everything we build, but if there's no foreseeable use for it (which I think applies here) it's wasted effort.  I think we could adapt to this as we need it in future.  Nice to be thinking about though!  I don't think I've ever seen anything other than a v4 UUID used in Sanger.

As a more useful generalisation, perhaps the function could be `_check_can_transform` and provide a function/lambda to the check method.  The check method just checks that nothing goes wrong when applying the transform function and raises a more general "value is not transformable / of the correct type".  I feel you're lessening the power of your error reporting though since it wouldn't know how to describe what the value was supposed to be.

Reply:
Fair enough. If any other versions aren't going be used there's no point of validating UUIDs for specific versions. Agreed.